include/deal.II/dofs/dof_tools.h

```diff
@@ -2258,7 +2258,7 @@ namespace DoFTools
    */
   template <typename DoFHandlerType>
   std::vector<types::global_dof_index>
-  count_dofs_per_component(
+  count_dofs_per_fe_component(
```

Comment:
I don't think that we have used it in deal.II so far, but this would be a great candidate for [[nodiscard]] to make the conversion from the old function harder to get wrong.

Reply:
Yes, nice idea. We'd need to write a bit of cmake detection code because the attribute is only C++17: https://en.cppreference.com/w/cpp/language/attributes/nodiscard

(That's a nice and drastic example in that page, by the way...)

Reply:
Could we leave that for a later patch?

Reply:
On second though: There are a great many functions that return something. In essence, every function that is `const` falls in this category, as well as all global functions with only `const` arguments. Unless we have a plan to mark *all* of these functions, I think doing it for this one here isn't making much of a point.

Reply:
We already check for `[[nodiscard]]` when we check for C++17. I thought about using it everywhere as well and there is a `clang-tidy` check (https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-nodiscard.html) for it. However, it results in some many code changes (with annoyingly long macro names) that I refrained from doing it.

Reply:
I am fine with doing it at some other point (or not at all).